packages/core/src/components/forms/numericInput.tsx

```diff
@@ -477,22 +483,26 @@ export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInp
     }
 
     private invokeOnValueChangeCallback(value: string) {
-        const valueAsString = value;
-        const valueAsNumber = +value; // coerces non-numeric strings to NaN
```

Comment:
can we keep this comment around? perhaps in `invokeOnValueChangeCallback`, to remind us why we're not using `parseInt`.